### PR TITLE
notification on incoming changes

### PR DESCRIPTION
**Pull request by norbusan**
This pull request implements notifications when changes are incoming from the cloud.
A new configuration option `min_notif_changes` has been added, controlling the minimum
number of incoming changes to trigger a desktop notification. If there are less than this
number, no notification will be shown.

### CODE REVIEW

**Comment by lhorace** [comment]
> Concerning rewrite: I don't know what you are talking about - I don't think any one of us has time and energy to do a complete rewrite.

You didn't know? Why issues are tagged future planning?

**Comment by lhorace** [comment]
See https://github.com/abraunegg/onedrive/issues/313#issuecomment-449870423

**Comment by norbusan** [comment]
Future planning and complee rewrite are different things. And your comment I have seen, but I disagree.

**Comment by lhorace** [comment]
Okay, I just don't understand why not voice it? Anyway, if you guys are not going to do it, I won't expect it nor bother myself worrying about it.

**Comment by abraunegg** [comment]
@lhorace 
> You didn't know? Why issues are tagged future planning?

I tag feature requests as 'Future Work - Planning Required' for the simple reason of:
* Requires more thought / white-boarding than I have time for
* Will most likely need some sort of coding & development / change to the code - ie more than 5 mins worth of time & effort

Re a 'complete re-write':

Some aspects of this currently client & how it is fundamentally written, the way it uses classes & functions today does not lend itself well to multi-threading activities. Being able to multi-thread correctly would mean being able to handle parallel uploads / download thus potentially a faster end user experience than today - esp when dealing with many small files.

Certain sections of the client would need to be re-written to support doing this as I have investigated a number of options here on how to do it - it is not a complete re-write per say - but would be a major engineering & testing effort and @norbusan is correct - this is something that there has to be time & energy for.

**Comment by lhorace** [comment]
Since it's a compile in behavior and you guys worked to make sure it works right. I'll leave it at that

File: src/sync.d [hunk 1]
```diff
@@ -579,8 +580,15 @@ final class SyncEngine
 			
 			// Are there any changes to process?
 			if (("value" in changes) != null) {
-				// There are valid changes
-				log.vdebug("Number of changes from OneDrive to process: ", count(changes["value"].array));
+				auto nrChanges = count(changes["value"].array);
+
+
```
Comment (abraunegg) [inline src/sync.d]: un-needed extra line here

**Comment by abraunegg** [comment]
Need to build & test .. will do this sometime later today

File: src/sync.d (hunk 1 above)
Comment (norbusan) [reply src/sync.d]: removed one empty line there. I still like to have one remaining between the var def and the rest.

**Comment by abraunegg** [comment]
@norbusan 
Currently testing this PR, found a new issue - https://github.com/abraunegg/onedrive/issues/365

**Comment by abraunegg** [comment]
@norbusan 
Changes look good to me & work well

**Comment by norbusan** [comment]
Thanks for merging

**Pull request merged**